include/unifex/let_error.hpp

```diff
@@ -252,68 +265,108 @@ class _op<Source, Error, Receiver>::type {
 
   using source_op_t = connect_result_t<Source, source_receiver>;
 
-  using final_sender_t = callable_result_t<Error>;
+  template <typename Error>
+  using final_sender_t = callable_result_t<Func&, remove_cvref_t<Error>>;
+
+  using final_receiver_t = final_receiver_type<Source, Func, Receiver>;
+
+  template <typename Error>
+  using final_op_t = connect_result_t<
+      final_sender_t<Error>,
+      final_receiver_t>;
+
+  template <typename... Errors>
+  using final_op_union = manual_lifetime_union<final_op_t<Errors>...>;
 
-  using final_op_t = connect_result_t<final_sender_t, final_receiver>;
+  using final_op_union_t = typename remove_cvref_t<Source>::template error_types<final_op_union>;
 
-  UNIFEX_NO_UNIQUE_ADDRESS Error error_;
+  UNIFEX_NO_UNIQUE_ADDRESS Func error_;
   UNIFEX_NO_UNIQUE_ADDRESS Receiver receiver_;
   state startedOp_ = state::neither;
   union {
     manual_lifetime<source_op_t> sourceOp_;
-    manual_lifetime<final_op_t> finalOp_;
+    final_op_union_t finalOp_;
   };
+  using deactivate_t = void(*)(final_op_union_t&) noexcept;
+  deactivate_t deactivate_;
 };
 
-template <typename Source, typename Error>
-class _sndr<Source, Error>::type {
-  using final_sender_t = callable_result_t<Error>;
+template<typename Sender>
+struct sends_done_impl : std::bool_constant<sender_traits<Sender>::sends_done> {};
+
+template<typename... Senders>
+using any_sends_done = std::disjunction<sends_done_impl<Senders>...>;
+
+template <typename Source, typename Func>
+class _sndr<Source, Func>::type {
+
+  template <typename Error>
+  using final_sender = callable_result_t<Func&, remove_cvref_t<Error>>;
+
+  template <template <typename...> class Tuple>
+  struct make_value_type_list {
+    template<typename... Errors>
+    using make = typename concat_type_lists_unique<
+      sender_value_types_t<final_sender<Errors>, type_list, Tuple>...>::type;
```

Comment:
Be careful using `concate_type_lists_unique` when passing the user-provided `Tuple` or `Variant` as elements.
e.g. what if tuple was `template<typename...> Tuple = void;`

Usually I build everything uniquely using `type_list` and then use the `type_list_nested_apply_t` helper to apply the user-provided `Tuple` and `Variant` versions.